### PR TITLE
Removes ts-option in favor of fp-ts

### PR DESCRIPTION
**Pull request by cloudify**

### CODE REVIEW

File: lib/controllers/messages.ts [hunk 1]
```diff
@@ -442,8 +443,8 @@ export function GetMessageHandler(
 
     const messageStatus: MessageResponse = {
       message: retrievedMessageToPublic(retrievedMessage),
-      notification: maybeNotificationStatus.isDefined
-        ? maybeNotificationStatus.get
+      notification: maybeNotificationStatus.isSome()
```
Comment (gunzip) [inline lib/controllers/messages.ts]: maybeNotificationStatus.toUndefined() without the isSome() check should suffice

File: lib/utils/env.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * Helper function that validates an environment variable and return its value
+ * if it's a `NonEmptyString`.
+ * Throws an Error otherwise.
+ */
+export function getRequiredStringEnv(k: string): NonEmptyString {
+  const maybeValue = toNonEmptyString(process.env[k]);
+
+  if (isNone(maybeValue)) {
+    throw new Error(`${k} must be defined and non-empty`);
```
Comment (gunzip) [inline lib/utils/env.ts]: maybe a call to winston log() here would be useful as well ?

File: lib/utils/env.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { isNone } from "fp-ts/lib/Option";
+import { NonEmptyString, toNonEmptyString } from "./strings";
+
+/**
+ * Helper function that validates an environment variable and return its value
+ * if it's a `NonEmptyString`.
+ * Throws an Error otherwise.
+ */
+export function getRequiredStringEnv(k: string): NonEmptyString {
```
Comment (gunzip) [inline lib/utils/env.ts]: better name getRequiredEnvString ?

File: lib/utils/middlewares/azure_api_auth.ts [hunk 4]
```diff
@@ -81,8 +81,8 @@ function getGroupsFromHeader(groupsHeader: string): Set<UserGroup> {
     groupsHeader
       .split(",")
       .map(v => toUserGroup(v))
-      .filter(g => g.isDefined)
-      .map(g => g.get)
+      .filter(g => isSome(g))
+      .map(g => (g as Some<UserGroup>).value)
```
Comment (gunzip) [inline lib/utils/middlewares/azure_api_auth.ts]: maybe typescript can infer the type using objects methods ie.  g.isSome() instead of isSome(g) ?

Comment (cloudify) [reply lib/utils/middlewares/azure_api_auth.ts]: unfortunately it does not

File: lib/utils/env.ts (hunk 2 above)
Comment (cloudify) [reply lib/utils/env.ts]: the exception should go into the logs already, doesn't it?

**Pull request merged**